cs301-itsa-ay2324/admin-ui-proxy
# feat: user role integration

## Overview
This PR includes changes for user role integration. According to project requirements, high-level admin should be able to perform CRUD operations on roles and permissions.

## Features
- [x] Role and Permissions
> - [x] Create a role with permissions
> - [x] View role
> - [x] Update role and its permissions
> - [x] Delete role
- [x] User
> - [x] When updating user details, new roles will be shown

## Testing
To carry out testing for this PR, please follow the steps below:

1. Ensure backend services (Docker) are up and running
2. Ensure all data exists in databases
3. Set up the project
```
yarn install
yarn dev
```
